### PR TITLE
Cldc 497 shared accomodation bedrooms

### PR DESCRIPTION
**Pull request by magicmilo**

### CODE REVIEW

File: app/models/case_log.rb [hunk 1]
```diff
@@ -143,6 +163,15 @@ def women_of_child_bearing_age_in_household(record)
       record["person_#{n}_gender"] == "Female" && record["person_#{n}_age"] >= 16 && record["person_#{n}_age"] <= 50
     end
   end
+
+  def people_in_household(record)
```
Comment (baarkerlounger) [inline app/models/case_log.rb]: Why would we need to derive this? Can't we just check `record.household_number_of_other_members`?

Comment (magicmilo) [reply app/models/case_log.rb]: Good spot thanks, switched it. Don't think the test should be dependent on this field being filled.

File: app/models/case_log.rb [hunk 2]
```diff
@@ -1,10 +1,136 @@
 class CaseLogValidator < ActiveModel::Validator
+  # Methods to be used on save and continue need to be named 'validate_'
+  # followed by field name this is how the metaprogramming of the method
+  # name being call in the validate method works.
+
+  def validate_person_1_age(record)
```
Comment (baarkerlounger) [inline app/models/case_log.rb]: This looks like a merge conflict. Think you want to check out this file from `main` and then put all your new validation methods in `/app/validators/household_validations.rb`

File: app/models/case_log.rb [hunk 3]
```diff
@@ -1,10 +1,136 @@
...
 
+  def validate_shared_housing_rooms(record)
+    unless record.property_unit_type.nil?
+      if record.property_unit_type == "Bed-sit" && record.property_number_of_bedrooms != 1
+        record.errors.add :property_unit_type, "A bedsit can only have one bedroom"
+      end
+
+      unless record.household_number_of_other_members.nil?
+        if record.household_number_of_other_members > 0
+          if record.property_unit_type.include?("Shared") && (record.property_number_of_bedrooms.to_i == 0 || record.property_number_of_bedrooms.to_i > 7)
```
Comment (baarkerlounger) [inline app/models/case_log.rb]: How about `record.property_number_of_bedrooms.between?(0, 7)`

Comment (magicmilo) [reply app/models/case_log.rb]: went with !record.property_number_of_bedrooms.to_i.between?(1, 7)

File: app/models/case_log.rb [hunk 4]
```diff
@@ -1,4 +1,53 @@
 class CaseLogValidator < ActiveModel::Validator
+  # Methods to be used on save and continue need to be named 'validate_'
+  # followed by field name this is how the metaprogramming of the method
+  # name being call in the validate method works.
+  def validate_property_number_of_times_relet(record)
```
Comment (baarkerlounger) [inline app/models/case_log.rb]: This looks like it's still coming from merge? Don't think this file needs to change at all?

**Review by baarkerlounger — APPROVED** [review]

**Pull request merged**